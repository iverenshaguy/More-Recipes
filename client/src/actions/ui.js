import { TOGGLE_MODAL, TOGGLE_REVIEW_FORM, CHANGE_RECIPE_FORM_STATE } from './actionTypes';

const toggleModal = modal => ({
  type: TOGGLE_MODAL,
  payload: modal || null
});


const toggleReviewForm = () => ({
  type: TOGGLE_REVIEW_FORM
});

const changeRecipeFormState = payload => ({
  type: CHANGE_RECIPE_FORM_STATE,
  payload
});

export default {
  toggleModal,
  toggleReviewForm,
  changeRecipeFormState
};
