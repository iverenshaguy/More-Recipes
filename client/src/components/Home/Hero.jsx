import React from 'react';
import { InputGroup, InputGroupButton, Button, Form, Label, Input } from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';

const Hero = props => (
  <div className="container-fluid" id="hero-image">
    <div className="row mx-auto" id="hero-image-text">
      <div className="col align-self-center">
        <div className="row justify-content-center align-items-center mb-3">
          <div className="col-12">
            <h3>Search for a Recipe</h3>
          </div>
          <div className="col-8 py-4">
            <Form id="search-form">
              <InputGroup size="lg">
                <Label className="sr-only" for="search">
                  Search for Recipes
                </Label>
                <Input id="search" placeholder="Curried Chicken Gravy..." />
                <InputGroupButton>
                  <Button type="submit" className="search-btn">
                    <FontAwesome name="search" />
                  </Button>
                </InputGroupButton>
              </InputGroup>
            </Form>
          </div>
        </div>
        <Button
          onClick={() => props.toggle('addRecipeModal')}
          className="btn-default btn-lg"
          id="home-add-recipe-btn"
          title="New Recipe"
        >
          Add a New Recipe
        </Button>
      </div>
    </div>
  </div>
);

Hero.propTypes = {
  toggle: PropTypes.func.isRequired,
};

export default Hero;
