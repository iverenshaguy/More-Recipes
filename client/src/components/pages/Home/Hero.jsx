import React from 'react';
import PropTypes from 'prop-types';
import { AddRecipeBtn } from '../../shared/Buttons';
import SearchForm from '../../shared/Forms/SearchForm';

/**
 * @exports
 * @function Hero
 * @param {object} props
 * @returns {JSX} Hero
 */
const Hero = (props) => {
  const {
    handleSearch, handleAddRecipe, handleSearchInput, searchValue
  } = props;

  return (
    <div className="container-fluid" id="hero-image">
      <div className="row mx-auto" id="hero-image-text">
        <div className="col align-self-center">
          <div className="row justify-content-center align-items-center mb-3">
            <div className="col-12">
              <h3>Search for a Recipe</h3>
            </div>
            <div className="col-8 py-4">
              <SearchForm
                size="lg"
                handleSearch={handleSearch}
                handleSearchInput={handleSearchInput}
                searchValue={searchValue}
              />
            </div>
          </div>
          <AddRecipeBtn handleClick={handleAddRecipe} />
        </div>
      </div>
    </div>);
};

Hero.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  handleAddRecipe: PropTypes.func.isRequired,
  handleSearchInput: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired
};

export default Hero;
