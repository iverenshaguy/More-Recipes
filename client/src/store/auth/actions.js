import {
  AUTHENTICATED,
  UNAUTHENTICATED,
  AUTHENTICATION_ERROR,
  AUTHENTICATING,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  SIGNUP_SUCCESS,
  SIGNUP_ERROR,
  CLEAR_AUTH_ERROR
} from './types';

const authenticating = () => ({
  type: AUTHENTICATING
});

const authenticationSuccess = data => ({
  type: AUTHENTICATED,
  payload: data
});

const authenticationFailure = error => ({
  type: AUTHENTICATION_ERROR,
  payload: error
});

const loginSuccess = data => ({
  type: LOGIN_SUCCESS,
  payload: data
});

const loginFailure = error => ({
  type: LOGIN_ERROR,
  payload: error
});

const signupSuccess = data => ({
  type: SIGNUP_SUCCESS,
  payload: data
});

const signupFailure = error => ({
  type: SIGNUP_ERROR,
  payload: error
});

const clearAuthError = () => ({
  type: CLEAR_AUTH_ERROR
});

const resetUser = () => ({
  type: UNAUTHENTICATED
});

export default {
  authenticating,
  authenticationSuccess,
  authenticationFailure,
  loginSuccess,
  loginFailure,
  signupSuccess,
  signupFailure,
  clearAuthError,
  resetUser
};
