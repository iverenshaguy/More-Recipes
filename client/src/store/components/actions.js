import { TOGGLE_MODAL } from './types';

const toggleModal = modal => ({
  type: TOGGLE_MODAL,
  payload: modal || ''
});

export default {
  toggleModal
};
