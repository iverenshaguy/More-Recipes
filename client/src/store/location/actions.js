import { HOME, AUTH } from './types';

const setHomeLocation = () => ({
  type: HOME
});

const setAuthLocation = () => ({
  type: AUTH
});

export default {
  setHomeLocation,
  setAuthLocation
};
