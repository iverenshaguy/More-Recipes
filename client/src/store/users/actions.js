import { AUTHENTICATED, AUTHENTICATION_ERROR } from './types';

const authenticationSuccess = () => ({
  type: AUTHENTICATED
});

const authenticationFailure = () => ({
  type: AUTHENTICATION_ERROR,
  payload: 'Invalid Email or Password'
});

export default {
  authenticationSuccess,
  authenticationFailure
};
