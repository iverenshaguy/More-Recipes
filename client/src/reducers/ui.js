import { TOGGLE_MODAL, TOGGLE_REVIEW_FORM } from '../actions/actionTypes';

const initialState = {
  modals: {
    isOpen: false,
    type: null
  },
  reviewForm: {
    isOpen: false
  }
};

export default (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_MODAL:
      return Object.assign({}, state, {
        modals: {
          isOpen: !state.modals.isOpen,
          type: action.payload || null
        }
      });
    case TOGGLE_REVIEW_FORM:
      return Object.assign({}, state, {
        reviewForm: {
          isOpen: !state.reviewForm.isOpen,
        }
      });
    default:
      return state;
  }
};
