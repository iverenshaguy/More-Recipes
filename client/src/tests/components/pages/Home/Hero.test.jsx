import React from 'react';
import Hero from '../../../../components/pages/Home/Hero';
import { toggleModal } from '../../../../actions/ui';

const setup = () => {
  const dispatch = jest.fn();

  const props = {
    handleSearchInput: jest.fn(),
    handleSearch: jest.fn(),
    searchValue: '',
    dispatch,
  };

  const shallowWrapper = shallow(<Hero {...props} />);
  const mountedWrapper = mount(<Hero {...props} />);

  return {
    shallowWrapper,
    mountedWrapper,
    dispatch,
  };
};

describe('Home: Hero', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { shallowWrapper } = setup();
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly when mouted fully', () => {
    const { mountedWrapper } = setup();
    expect(toJson(mountedWrapper)).toMatchSnapshot();

    mountedWrapper.unmount();
  });

  it('calls toggleModal method on add recipe button click', () => {
    const { shallowWrapper, dispatch } = setup();

    shallowWrapper.find('#home-add-recipe-btn').simulate('click');
    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(dispatch).toHaveBeenCalledWith(toggleModal('addRecipe'));
  });
});
