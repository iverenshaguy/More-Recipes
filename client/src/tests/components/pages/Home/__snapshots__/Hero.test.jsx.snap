// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Hero renders correctly 1`] = `
<div
  className="container-fluid"
  id="hero-image"
>
  <div
    className="row mx-auto"
    id="hero-image-text"
  >
    <div
      className="col align-self-center"
    >
      <div
        className="row justify-content-center align-items-center mb-3"
      >
        <div
          className="col-12"
        >
          <h3>
            Search for a Recipe
          </h3>
        </div>
        <div
          className="col-8 py-4"
        >
          <SearchForm
            handleSearch={[Function]}
            handleSearchInput={[Function]}
            searchValue=""
            size="lg"
          />
        </div>
      </div>
      <Button
        className="btn-default btn-lg"
        color="secondary"
        id="home-add-recipe-btn"
        onClick={[Function]}
        tag="button"
        title="New Recipe"
      >
        Add a New Recipe
      </Button>
    </div>
  </div>
</div>
`;

exports[`Home: Hero renders correctly when mouted fully 1`] = `
<Hero
  dispatch={[Function]}
  handleSearch={[Function]}
  handleSearchInput={[Function]}
  searchValue=""
>
  <div
    className="container-fluid"
    id="hero-image"
  >
    <div
      className="row mx-auto"
      id="hero-image-text"
    >
      <div
        className="col align-self-center"
      >
        <div
          className="row justify-content-center align-items-center mb-3"
        >
          <div
            className="col-12"
          >
            <h3>
              Search for a Recipe
            </h3>
          </div>
          <div
            className="col-8 py-4"
          >
            <SearchForm
              handleSearch={[Function]}
              handleSearchInput={[Function]}
              searchValue=""
              size="lg"
            >
              <Form
                id="search-form"
                onSubmit={[Function]}
                tag="form"
              >
                <form
                  className=""
                  id="search-form"
                  onSubmit={[Function]}
                >
                  <InputGroup
                    size="lg"
                    tag="div"
                  >
                    <div
                      className="input-group input-group-lg"
                    >
                      <Label
                        className="sr-only"
                        for="search"
                        tag="label"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <label
                          className="sr-only  form-control-label"
                          htmlFor="search"
                        >
                          Search for Recipes
                        </label>
                      </Label>
                      <Input
                        id="search"
                        onChange={[Function]}
                        placeholder="Curried Chicken Gravy..."
                        tag="p"
                        type="text"
                        value=""
                      >
                        <input
                          className="form-control"
                          id="search"
                          onChange={[Function]}
                          placeholder="Curried Chicken Gravy..."
                          type="text"
                          value=""
                        />
                      </Input>
                      <InputGroupButton
                        tag="div"
                      >
                        <div
                          className="input-group-btn"
                        >
                          <Button
                            className="search-btn"
                            color="secondary"
                            tag="button"
                            type="submit"
                          >
                            <button
                              className="search-btn btn btn-secondary"
                              onClick={[Function]}
                              type="submit"
                            >
                              <FontAwesome
                                name="search"
                              >
                                <span
                                  aria-hidden={true}
                                  className="fa fa-search"
                                />
                              </FontAwesome>
                            </button>
                          </Button>
                        </div>
                      </InputGroupButton>
                    </div>
                  </InputGroup>
                </form>
              </Form>
            </SearchForm>
          </div>
        </div>
        <Button
          className="btn-default btn-lg"
          color="secondary"
          id="home-add-recipe-btn"
          onClick={[Function]}
          tag="button"
          title="New Recipe"
        >
          <button
            className="btn-default btn-lg btn btn-secondary"
            id="home-add-recipe-btn"
            onClick={[Function]}
            title="New Recipe"
            type="button"
          >
            Add a New Recipe
          </button>
        </Button>
      </div>
    </div>
  </div>
</Hero>
`;
